shell/common/common_TPC-H.sh

```diff
@@ -1,40 +1,73 @@
-
-BENCH_DATA_SIZE="$((TPCH_SCALE_FACTOR * 1000000000 ))" #in bytes
-
+#
+if [[ "$NATIVE_FORMAT" == "text" && "$BENCH_SUITE" == *"native-spark"* ]]; then
+  source_file "$ALOJA_REPO_PATH/shell/common/common_hadoop.sh"
+  set_hadoop_requires
+else
+  source_file "$ALOJA_REPO_PATH/shell/common/common_hive.sh"
+  set_hive_requires
+fi
+
+if [ "$BB_SERVER_DERBY" == "true" ]; then
+  source_file "$ALOJA_REPO_PATH/shell/common/common_derby.sh"
+  set_derby_requires
+fi
+
+
+if [[ "$TPCH_USE_LOCAL_FACTOR" > 0 ]]; then
+  TPCH_SCALE_FACTOR=$TPCH_USE_LOCAL_FACTOR
+  BENCH_DATA_SIZE="$((TPCH_USE_LOCAL_FACTOR * 1000000000 ))" #in bytes
+else
+  [[ ! "$TPCH_SCALE_FACTOR" ]] && TPCH_SCALE_FACTOR=1 #1 GB min size
+  BENCH_DATA_SIZE="$((TPCH_SCALE_FACTOR * 1000000000 ))" #in bytes
+fi
 
 TPCH_HDFS_DIR="/tmp/tpch-generate"
-TPCH_DB_NAME="tpch_${BENCH_FILE_FORMAT}_${TPCH_SCALE_FACTOR}"
 
-[ ! "$BENCH_LIST" ] && BENCH_LIST="$(seq -f "tpch_query%g" -s " " 1 22)"
+if [[ ! "$NATIVE_FORMAT" == "text"  || ! "$BENCH_SUITE" == *"native-spark"* ]]; then
+  [ ! -z $NATIVE_FORMAT ] && BENCH_FILE_FORMAT=$NATIVE_FORMAT
+  TPCH_DB_NAME="tpch_${BENCH_FILE_FORMAT}_${TPCH_SCALE_FACTOR}"
+fi
+
+[ ! "$BENCH_LIST" ] && BENCH_LIST="$(printf '%d ' {1..22})"
 
 # Validations
 [ "$(get_hadoop_major_version)" != "2" ] && die "Hadoop v2 is required for $BENCH_SUITE"
-[ "$BENCH_FILE_FORMAT" != "orc" ] && die "Only orc file format is supported for now, got: $BENCH_FILE_FORMAT"
+[[ "$BENCH_FILE_FORMAT" != "orc" && ! "$BENCH_SUITE" == *"native-spark"* ]] && die "Only orc file format is supported for now, got: $BENCH_FILE_FORMAT"
 
 # TODO: temporary patch for missing gcc on azure ubuntu
 [ ! "$(which gcc)" ] && sudo apt-get install -y -q gcc make
 [ ! "$(which gcc)" ] && die "Build tools not installed for TPC-H datagen to work"
 
 D2F_folder_name="D2F-Bench-master"
-BENCH_REQUIRED_FILES["$D2F_folder_name"]="http://github.com/Aloja/D2F-Bench/archive/master.zip"
+BENCH_REQUIRED_FILES["$D2F_folder_name"]="https://github.com/Aloja/D2F-Bench/archive/master.zip"
 D2F_local_dir="$(get_local_apps_path)/$D2F_folder_name"
 
-
 benchmark_suite_config() {
   initialize_hadoop_vars
   prepare_hadoop_config "$NET" "$DISK" "$BENCH_SUITE"
   start_hadoop
 
-  initialize_hive_vars
-  prepare_hive_config "$HIVE_SETTINGS_FILE" "$HIVE_SETTINGS_FILE_PATH"
+  # Set hive for ORC
+  if [[ ! "$NATIVE_FORMAT" == "text" || ! "$BENCH_SUITE" =~ *"native-spark"* ]]; then
+    initialize_hive_vars
+    prepare_hive_config "$HIVE_SETTINGS_FILE" "$HIVE_SETTINGS_FILE_PATH"
+      if [ "$BB_SERVER_DERBY" == "true" ]; then
+        logger "WARNING: Using Derby DB in client/server mode"
+        USE_EXTERNAL_DATABASE="true"
+        initialize_derby_vars "TPCH_DB"
+        start_derby
+      else
+        logger "WARNING: Using Derby DB in embedded mode"
+      fi
+  fi
+
+
+  if [ ! -z "$use_spark" ]; then
```

Comment:
So we already have the `use_spark` variable here? I thought that was something new in the other branch.

If we are already using this variable, my previous comments about the way this is written still stand - we should check against a specific value (e.g. `true` or `1`).

Reply:
Use_$engine was used prior to the new branch. The functionality is basically the same.